From f5f6c46dec0d39cd0986ecb6b1421b45c3b73aa9 Mon Sep 17 00:00:00 2001
From: Andrew Purtell <apurtell@apache.org>
Date: Fri, 3 Mar 2017 12:31:00 -0800
Subject: [PATCH 330/362] HBASE-17722 Metrics subsystem stop/start messages
 add a lot of useless bulk to operational logging

Ref: CDH-53383
Reviewed By: Busbey

Conflicts:
	hbase-spark/src/test/resources/log4j.properties

Conflicts:
	conf/log4j.properties
	hbase-archetypes/hbase-client-project/src/main/resources/log4j.properties
	hbase-archetypes/hbase-shaded-client-project/src/main/resources/log4j.properties

(cherry picked from commit 5ab33b3cd77d6e1946fea00c9e8e75a5836e8184)

Change-Id: Icd6cb2eaacfb3e8d5803284aecf09754fd9caa62
---
 conf/log4j.properties                              |    5 +++++
 hbase-client/src/test/resources/log4j.properties   |    8 +++++---
 hbase-common/src/test/resources/log4j.properties   |    8 +++++---
 hbase-examples/src/test/resources/log4j.properties |    8 +++++---
 .../src/test/resources/log4j.properties            |    8 +++++---
 .../src/test/resources/log4j.properties            |    8 +++++---
 hbase-rest/src/test/resources/log4j.properties     |    8 +++++---
 hbase-server/src/test/resources/log4j.properties   |    8 +++++---
 hbase-shell/src/test/resources/log4j.properties    |    8 +++++---
 9 files changed, 45 insertions(+), 24 deletions(-)

diff --git a/conf/log4j.properties b/conf/log4j.properties
index a15d575..2e482cd 100644
--- a/conf/log4j.properties
+++ b/conf/log4j.properties
@@ -108,3 +108,8 @@ log4j.logger.org.apache.hadoop.hbase.zookeeper.ZooKeeperWatcher=INFO
 # and scan of hbase:meta messages
 # log4j.logger.org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation=INFO
 # log4j.logger.org.apache.hadoop.hbase.client.MetaScanner=INFO
+
+# Prevent metrics subsystem start/stop messages (HBASE-17722)
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsConfig=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSinkAdapter=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=WARN
diff --git a/hbase-client/src/test/resources/log4j.properties b/hbase-client/src/test/resources/log4j.properties
index 56f8f92..405435e 100644
--- a/hbase-client/src/test/resources/log4j.properties
+++ b/hbase-client/src/test/resources/log4j.properties
@@ -58,9 +58,11 @@ log4j.logger.org.apache.hadoop=WARN
 log4j.logger.org.apache.zookeeper=ERROR
 log4j.logger.org.apache.hadoop.hbase=DEBUG
 
-#These two settings are workarounds against spurious logs from the minicluster.
+#These settings are workarounds against spurious logs from the minicluster.
 #See HBASE-4709
-log4j.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=ERROR
-log4j.org.apache.hadoop.metrics2.util.MBeans=ERROR
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsConfig=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSinkAdapter=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=WARN
+log4j.logger.org.apache.hadoop.metrics2.util.MBeans=WARN
 # Enable this to get detailed connection error/retry logging.
 # log4j.logger.org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation=TRACE
diff --git a/hbase-common/src/test/resources/log4j.properties b/hbase-common/src/test/resources/log4j.properties
index 56f8f92..405435e 100644
--- a/hbase-common/src/test/resources/log4j.properties
+++ b/hbase-common/src/test/resources/log4j.properties
@@ -58,9 +58,11 @@ log4j.logger.org.apache.hadoop=WARN
 log4j.logger.org.apache.zookeeper=ERROR
 log4j.logger.org.apache.hadoop.hbase=DEBUG
 
-#These two settings are workarounds against spurious logs from the minicluster.
+#These settings are workarounds against spurious logs from the minicluster.
 #See HBASE-4709
-log4j.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=ERROR
-log4j.org.apache.hadoop.metrics2.util.MBeans=ERROR
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsConfig=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSinkAdapter=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=WARN
+log4j.logger.org.apache.hadoop.metrics2.util.MBeans=WARN
 # Enable this to get detailed connection error/retry logging.
 # log4j.logger.org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation=TRACE
diff --git a/hbase-examples/src/test/resources/log4j.properties b/hbase-examples/src/test/resources/log4j.properties
index 4eeeb2c..c322699 100644
--- a/hbase-examples/src/test/resources/log4j.properties
+++ b/hbase-examples/src/test/resources/log4j.properties
@@ -58,9 +58,11 @@ log4j.logger.org.apache.hadoop=WARN
 log4j.logger.org.apache.zookeeper=ERROR
 log4j.logger.org.apache.hadoop.hbase=DEBUG
 
-#These two settings are workarounds against spurious logs from the minicluster.
+#These settings are workarounds against spurious logs from the minicluster.
 #See HBASE-4709
-log4j.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=ERROR
-log4j.org.apache.hadoop.metrics2.util.MBeans=ERROR
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsConfig=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSinkAdapter=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=WARN
+log4j.logger.org.apache.hadoop.metrics2.util.MBeans=WARN
 # Enable this to get detailed connection error/retry logging.
 # log4j.logger.org.apache.hadoop.hbase.client.ConnectionImplementation=TRACE
diff --git a/hbase-prefix-tree/src/test/resources/log4j.properties b/hbase-prefix-tree/src/test/resources/log4j.properties
index 4eeeb2c..c322699 100644
--- a/hbase-prefix-tree/src/test/resources/log4j.properties
+++ b/hbase-prefix-tree/src/test/resources/log4j.properties
@@ -58,9 +58,11 @@ log4j.logger.org.apache.hadoop=WARN
 log4j.logger.org.apache.zookeeper=ERROR
 log4j.logger.org.apache.hadoop.hbase=DEBUG
 
-#These two settings are workarounds against spurious logs from the minicluster.
+#These settings are workarounds against spurious logs from the minicluster.
 #See HBASE-4709
-log4j.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=ERROR
-log4j.org.apache.hadoop.metrics2.util.MBeans=ERROR
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsConfig=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSinkAdapter=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=WARN
+log4j.logger.org.apache.hadoop.metrics2.util.MBeans=WARN
 # Enable this to get detailed connection error/retry logging.
 # log4j.logger.org.apache.hadoop.hbase.client.ConnectionImplementation=TRACE
diff --git a/hbase-procedure/src/test/resources/log4j.properties b/hbase-procedure/src/test/resources/log4j.properties
index 4eeeb2c..c322699 100644
--- a/hbase-procedure/src/test/resources/log4j.properties
+++ b/hbase-procedure/src/test/resources/log4j.properties
@@ -58,9 +58,11 @@ log4j.logger.org.apache.hadoop=WARN
 log4j.logger.org.apache.zookeeper=ERROR
 log4j.logger.org.apache.hadoop.hbase=DEBUG
 
-#These two settings are workarounds against spurious logs from the minicluster.
+#These settings are workarounds against spurious logs from the minicluster.
 #See HBASE-4709
-log4j.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=ERROR
-log4j.org.apache.hadoop.metrics2.util.MBeans=ERROR
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsConfig=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSinkAdapter=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=WARN
+log4j.logger.org.apache.hadoop.metrics2.util.MBeans=WARN
 # Enable this to get detailed connection error/retry logging.
 # log4j.logger.org.apache.hadoop.hbase.client.ConnectionImplementation=TRACE
diff --git a/hbase-rest/src/test/resources/log4j.properties b/hbase-rest/src/test/resources/log4j.properties
index 56f8f92..405435e 100644
--- a/hbase-rest/src/test/resources/log4j.properties
+++ b/hbase-rest/src/test/resources/log4j.properties
@@ -58,9 +58,11 @@ log4j.logger.org.apache.hadoop=WARN
 log4j.logger.org.apache.zookeeper=ERROR
 log4j.logger.org.apache.hadoop.hbase=DEBUG
 
-#These two settings are workarounds against spurious logs from the minicluster.
+#These settings are workarounds against spurious logs from the minicluster.
 #See HBASE-4709
-log4j.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=ERROR
-log4j.org.apache.hadoop.metrics2.util.MBeans=ERROR
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsConfig=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSinkAdapter=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=WARN
+log4j.logger.org.apache.hadoop.metrics2.util.MBeans=WARN
 # Enable this to get detailed connection error/retry logging.
 # log4j.logger.org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation=TRACE
diff --git a/hbase-server/src/test/resources/log4j.properties b/hbase-server/src/test/resources/log4j.properties
index 56f8f92..405435e 100644
--- a/hbase-server/src/test/resources/log4j.properties
+++ b/hbase-server/src/test/resources/log4j.properties
@@ -58,9 +58,11 @@ log4j.logger.org.apache.hadoop=WARN
 log4j.logger.org.apache.zookeeper=ERROR
 log4j.logger.org.apache.hadoop.hbase=DEBUG
 
-#These two settings are workarounds against spurious logs from the minicluster.
+#These settings are workarounds against spurious logs from the minicluster.
 #See HBASE-4709
-log4j.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=ERROR
-log4j.org.apache.hadoop.metrics2.util.MBeans=ERROR
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsConfig=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSinkAdapter=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=WARN
+log4j.logger.org.apache.hadoop.metrics2.util.MBeans=WARN
 # Enable this to get detailed connection error/retry logging.
 # log4j.logger.org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation=TRACE
diff --git a/hbase-shell/src/test/resources/log4j.properties b/hbase-shell/src/test/resources/log4j.properties
index 4eeeb2c..c322699 100644
--- a/hbase-shell/src/test/resources/log4j.properties
+++ b/hbase-shell/src/test/resources/log4j.properties
@@ -58,9 +58,11 @@ log4j.logger.org.apache.hadoop=WARN
 log4j.logger.org.apache.zookeeper=ERROR
 log4j.logger.org.apache.hadoop.hbase=DEBUG
 
-#These two settings are workarounds against spurious logs from the minicluster.
+#These settings are workarounds against spurious logs from the minicluster.
 #See HBASE-4709
-log4j.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=ERROR
-log4j.org.apache.hadoop.metrics2.util.MBeans=ERROR
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsConfig=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSinkAdapter=WARN
+log4j.logger.org.apache.hadoop.metrics2.impl.MetricsSystemImpl=WARN
+log4j.logger.org.apache.hadoop.metrics2.util.MBeans=WARN
 # Enable this to get detailed connection error/retry logging.
 # log4j.logger.org.apache.hadoop.hbase.client.ConnectionImplementation=TRACE
-- 
1.7.9.5

